backend/dal/PIMSContextFactory.cs

```diff
@@ -45,27 +45,44 @@ public PimsContextFactory()
         public PimsContext CreateDbContext(string[] args)
         {
             DotNetEnv.Env.Load();
-            string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+            string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Production";
 
-            var config = new ConfigurationBuilder()
+            // As per Microsoft documentation, a typical sequence of configuration providers is:
+            //   1. appsettings.json
+            //   2. appsettings.{Environment}.json
+            //   3. Secret Manager (if in development)
+            //   4. Environment variables using the Environment Variables configuration provider.
+            //   5. Command - line arguments using the Command-line configuration provider.
+            // source: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#configuration-providers
+            var builder = new ConfigurationBuilder()
                 .SetBasePath(Directory.GetCurrentDirectory())
-                .AddEnvironmentVariables()
-                .AddJsonFile("connectionstrings.json", optional : true, reloadOnChange : true)
-                .AddJsonFile($"connectionstrings.{environment}.json", optional : true, reloadOnChange : true)
-                .AddUserSecrets<PimsContext>()
-                .Build();
+                .AddJsonFile("connectionstrings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"connectionstrings.{environment}.json", optional: true, reloadOnChange: true);
+
+            if (!this.isProduction(environment))
```

Comment:
This isn’t absolutely required as the way secrets works wouldn’t even be relevant in the production environment.  But it doesn’t hurt to have the check.  However there is a built in extension method for the environment name that should be used rather than a custom one.

Reply:
I'm okay removing the check for "not Production" - just not sure whether a miss-configured app would be able to read from the secret json file in production.
All code I have seen always "wraps" the `UseSecrets` call with an `if (env.isDevelopment)` so it makes me wonder if secrets **don't work** in production why wrap them in an `if`?

Could it be that secret manager always writes to the json file and it's up to the developer to not use it in production? - or is it more like MSFT put in "smarts" into the secret manager so it returns `null` when in production? I really don't know what the guidance is...

All I see in the docs is "use Secret Manager in development and Azure Vault in Production". 

Again, I'm fine removing the check

Reply:
Nah leave it in.  It just wouldn't be relevant in a production environment due to all the other required elements (file/configuration).  No different than when the other optional files don't exist.